packages/kbn-server-route-repository/src/register_routes.test.ts

```diff
@@ -167,38 +177,227 @@ describe('registerRoutes', () => {
+    const error = await wrappedHandler(mockContext, mockRequest, kibanaResponseFactory);
+
+    expect(handler).toHaveBeenCalledTimes(1);
+    expect(error).toEqual({
       status: 500,
       payload: { message: 'error', attributes: { data: {} } },
       options: {},
     });
   });
+
+  describe('when using zod', () => {
+    const makeZodValidationObjectSpy = jest.spyOn(
+      makeZodValidationObject,
+      'makeZodValidationObject'
+    );
+
+    const zodParamsRt = z.object({
+      body: z.object({
+        bodyParam: z.string(),
+      }),
+      query: z.object({
+        queryParam: z.string(),
+      }),
+      path: z.object({
+        pathParam: z.string(),
+      }),
+    });
+
+    it('uses Core validation', () => {
+      callRegisterRoutes({
+        'POST /internal/route': {
+          endpoint: 'POST /internal/route',
+          params: zodParamsRt,
+          handler: jest.fn,
+        },
+      });
+
+      const [internalRoute] = post.mock.calls[0];
+      expect(makeZodValidationObjectSpy).toHaveBeenCalledWith(zodParamsRt);
+      expect(internalRoute.validate).toEqual(makeZodValidationObjectSpy.mock.results[0].value);
+    });
+
+    // params come in correctly, but bit odd since it's not being validated?
+    it('passes on params', async () => {
+      const handler = jest.fn();
+      callRegisterRoutes({
+        'POST /internal/route': {
+          endpoint: 'POST /internal/route',
+          params: zodParamsRt,
+          handler,
+        },
+      });
+
+      const [_, wrappedHandler] = post.mock.calls[0];
+
+      await wrappedHandler(
+        mockContext,
+        {
+          ...mockRequest,
+          params: {
+            pathParam: 'path',
+          },
+          query: {
+            queryParam: 'query',
+          },
+          body: {
+            bodyParam: 'body',
+          },
+        },
+        kibanaResponseFactory
+      );
+
+      expect(handler).toBeCalledTimes(1);
+      const [args] = handler.mock.calls[0];
+      const { params } = args;
+      expect(params).toEqual({
+        path: {
+          pathParam: 'path',
+        },
+        query: {
+          queryParam: 'query',
+        },
+        body: {
+          bodyParam: 'body',
+        },
+      });
+    });
+  });
+
+  describe('when using io-ts', () => {
+    const iotsParamsRt = t.type({
+      body: t.type({
+        bodyParam: t.string,
+      }),
+      query: t.type({
+        queryParam: t.string,
+      }),
+      path: t.type({
+        pathParam: t.string,
+      }),
+    });
+
+    it('bypasses Core validation', () => {
+      callRegisterRoutes({
+        'POST /internal/route': {
+          endpoint: 'POST /internal/route',
+          params: iotsParamsRt,
+          handler: jest.fn,
+        },
+      });
+
+      const [internalRoute] = post.mock.calls[0];
+      expect(internalRoute.validate).toEqual(passThroughValidationObject);
+    });
+
+    it('decodes params', async () => {
```

Comment:
I'm mostly checking data flow here, since we have tests for the strictness of decode request params